src/cmd/dataPatchEmissions.ts

```diff
@@ -0,0 +1,102 @@
+import ProgressBar from 'progress';
+import config from '../config';
+import { PoolClient } from 'pg';
+
+const batchSize = 10000;
+
+async function run(): Promise<void> {
+    const pool = await config.pg();
+
+    const client = await pool.connect();
+    try {
+        await client.query('BEGIN');
+
+        // Fetch all product hashes
+        config.logger.info('Fetching all product hashes');
+        const productHashes = await getProductHashes(client);
+
+        const progressBar = new ProgressBar(
+            '-> loading [:bar] :percent (:etas remaining)',
+            {
+              width: 40,
+              complete: '=',
+              incomplete: ' ',
+              renderThrottle: 500,
+              total: productHashes.length,
+            }
+          );
+
+        // // For each product hash, insert by batch the emissions data into the product table
+        // config.logger.info('Updating product emissions');
+        // for (let i = 0; i < productHashes.length; i += batchSize) {
+        //     const batch = productHashes.slice(i, i + batchSize);
+
+        //     await client.query('BEGIN');
+
+        //     await Promise.all(batch.map((productHash) => {
+        //         updateProductEmissions(client, productHash);
+        //         progressBar.tick();
+        //     }));
+
+        //     await client.query('COMMIT');
+        // }
+
+        config.logger.info('Updating product emissions...');
+        await updateProductsEmissions(client, productHashes);
+
+        // await setEmissionsUpdateSuccessful(client);
+
+        await client.query('COMMIT');
+    } catch (e) {
+        await client.query('ROLLBACK');
+
+        // await setEmissionsUpdateFailed(client);
+
+        throw e;
+    }
+
+}
+
+async function getProductHashes(client: PoolClient): Promise<string[]> {
+    const result = await client.query(
+        `
+        SELECT "productHash"
+        FROM "products"
+        `
+    );
+
+    return result.rows.map((row) => row.productHash);
+}
+
+async function updateProductsEmissions(client: PoolClient, productHashes: string[]): Promise<void> {
+    // Batch update emissions data for each product
+    await client.query(`
+        UPDATE "products"
+        SET "emissions" = $1
+        WHERE "productHash" = $2
+    `, [generateEmissionsData(), productHashes]);
+}
+
+const generateEmissionsData = () => {
+    return JSON.stringify([{
+        emissionHash: 'SampleEmissionHash',
+        unit: 'kgeqCO2',
```

Comment:
Just so that we keep it in mind: In the CLI we use "kgCO2e" (I don't know if changing it here implies making other changes somewhere else...)
```suggestion
        unit: 'kgCO2e',
```